lib/Models/Catalog/CatalogGroups/OpenDataSoftCatalogGroup.ts

```diff
@@ -227,7 +233,10 @@ export class OpenDataSoftCatalogStratum extends LoadableStratum(
   }
 
   getDatasetId(dataset: ValidDataset) {
-    return `${this.catalogGroup.uniqueId}/${dataset.dataset_id}`;
+    // Use OpenDataSoft server hostname for datasets, so we don't create multiple across facets
+    return `${URI(this.catalogGroup.url ?? "").hostname()}/${
+      dataset.dataset_id
+    }`;
```

Comment:
Should we fallback to parent group id when hostname is empty?

Reply:
Yeah I have added the fallback. Practically it will never be empty, but good to be safe